Dapper/SqlMapper.cs

```diff
@@ -3400,9 +3391,115 @@ private static LocalBuilder GetTempLocal(ILGenerator il, ref Dictionary<Type, Lo
                 il.Emit(OpCodes.Box, type);
             }
             il.Emit(OpCodes.Ret);
+        }
 
-            var funcType = System.Linq.Expressions.Expression.GetFuncType(typeof(IDataReader), returnType);
-            return (Func<IDataReader, object>)dm.CreateDelegate(funcType);
+        private static void LoadDefaultValue(ILGenerator il, Type type)
+        {
+            if (type.IsValueType())
+            {
+                int localIndex = il.DeclareLocal(type).LocalIndex;
+                LoadLocalAddress(il, localIndex);
+                il.Emit(OpCodes.Initobj, type);
+                LoadLocal(il, localIndex);
+            }
+            else
+            {
+                il.Emit(OpCodes.Ldnull);
+            }
+        }
+
+        private static void LoadReaderValueOrBranchToDBNullLabel(ILGenerator il, int index, ref int enumDeclareLocal, int? valueCopyLocal, Type colType, Type memberType, out Label isDbNullLabel)
+        {
+            isDbNullLabel = il.DefineLabel();
+            il.Emit(OpCodes.Ldarg_0); // stack is now [...][reader]
+            EmitInt32(il, index); // stack is now [...][reader][index]
+            il.Emit(OpCodes.Dup);// stack is now [...][reader][index][index]
+            il.Emit(OpCodes.Stloc_0);// stack is now [...][reader][index]

```

Comment:
This line introduced bug https://github.com/StackExchange/Dapper/issues/1279, fixed in https://github.com/StackExchange/Dapper/pull/1280.